e2etest/declarativeHelpers.go

```diff
@@ -0,0 +1,528 @@
+// Copyright © Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package e2etest
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Azure/azure-storage-azcopy/cmd"
+	"github.com/Azure/azure-storage-azcopy/common"
+	"github.com/JeffreyRichter/enum/enum"
+	"math"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+///////////
+
+// folder is syntactic sugar to make folders stand out in our file lists
+func folder(s string) string {
+	return strings.TrimRight(s, "/") + "/" // ensure it ends in one slash
```

Comment:
Minor: Might align with the 'f' file shortcut if this is 'd' for directory

Reply:
Maybe.  I'm hesitant because we've used the word "folder" so predominantly in the code. And also, because having them different lengths makes the folders look more distinct visually, which is not a bad thing IMHO, when it comes to reading and understanding the tests.

Reply:
I'll leave it as-is for now, but you could change it one day if that's the consensus opinion.